js_modules/dagster-ui/packages/ui-core/src/overview/OverviewAssetsRoot.tsx

```diff
@@ -240,13 +240,16 @@ function VirtualRow({height, start, group}: RowProps) {
     }
     Object.keys(liveDataByNode).forEach((key) => {
       const assetLiveData = liveDataByNode[key];
-      const asset = group.assets.find((asset) => toGraphId(asset.key) === key)!;
+      const asset = group.assets.find((asset) => toGraphId(asset.key) === key);
+      if (!asset) {
+        return;
+      }
       if (!asset.definition) {
         console.warn('Expected a definition for asset with key', key);
```

Comment:
Hmmm, conceptually I'm not sure definition should be optional in `AssetNodeStatusContent.tsx` - it seems like the two checks in there are fairly innocuous but it /should/ always exist and building status without it could return the wrong state.

Can you make this an early return instead of a `console.warn` and fix the problem that way maybe?

Reply:
Yeah, I was initially considering that but kept going back and forth. I'll change it back.